scvi/model/base/_archesmixin.py

```diff
@@ -155,6 +154,78 @@ def load_query_data(
 
         return model
 
+    @staticmethod
+    def prepare_query_anndata(
+        adata: AnnData,
+        reference_model: Union[str, BaseModelClass],
+        return_reference_var_names: bool = False,
+        inplace: bool = True,
+    ) -> Optional[Union[AnnData, pd.Index]]:
+        """
+        Prepare data for query integration.
+
+        This function will return a new AnnData object with padded zeros
+        for missing features, as well as correctly sorted features.
+
+        Parameters
+        ----------
+        adata
+            AnnData organized in the same way as data used to train model.
+            It is not necessary to run setup_anndata,
+            as AnnData is validated against the saved `scvi` setup dictionary.
+        reference_model
+            Either an already instantiated model of the same class, or a path to
+            saved outputs for reference model.
+        return_reference_var_names
+            Only load and return reference var names if True.
+        inplace
+            Whether to subset and rearrange query vars inplace or return new AnnData.
+
+        Returns
+        -------
+        Query adata ready to use in `load_query_data` unless `return_reference_var_names`
+        in which case a pd.Index of reference var names is returned.
+        """
+        _, var_names, _ = _get_loaded_data(reference_model)
+        var_names = pd.Index(var_names)
+
+        if return_reference_var_names:
```

Comment:
this is kind of strange sub sub functionality of this function. Just split into its own public function instead?

Reply:
I know... I didn't want to add another method for something so basic...

Reply:
though could just be a method called `load_var_names`